File: backend/src/models/quest.py
"""Quest 相关 Pydantic 模型 — 对齐 api-contracts.md。"""

from pydantic import BaseModel, Field


class QuestPreviewDayDTO(BaseModel):
    day_index: int
    event_title: str
    event_description: str
    estimated_time: str
    meaning: str
    output_type: str = "text"
    is_boss: bool = False


class QuestPreviewDTO(BaseModel):
    quest_title: str
    success_condition: str
    total_estimated_time: str
    days: list[QuestPreviewDayDTO]


class GenerateQuestRequest(BaseModel):
    goal_id: str


class GenerateQuestResponseData(BaseModel):
    preview: QuestPreviewDTO


class AcceptQuestPreviewInput(BaseModel):
    quest_title: str
    success_condition: str
    days: list[QuestPreviewDayDTO] = Field(default_factory=list)


class AcceptQuestRequest(BaseModel):
    goal_id: str
    preview: AcceptQuestPreviewInput | None = None


class QuestDTO(BaseModel):
    quest_id: str
    goal_id: str
    quest_title: str
    success_condition: str
    total_days: int
    current_day: int
    status: str
    created_at: str
    updated_at: str


class DailyEventDTO(BaseModel):
    event_id: str
    quest_id: str
    day_index: int
    event_title: str
    event_description: str
    estimated_time: str
    meaning: str
    output_type: str
    user_output: str | None
    status: str
    original_event_id: str | None
    completed_at: str | None


class AcceptQuestResponseData(BaseModel):
    quest: QuestDTO
    today_event: DailyEventDTO


class QuestProgressDTO(BaseModel):
    completed_days: int
    total_days: int


class ActiveQuestResponseData(BaseModel):
    quest: QuestDTO
    progress: QuestProgressDTO
    assets_count: int


class QuestDaySummaryDTO(BaseModel):
    day_index: int
    event_title: str
    estimated_time: str
    status: str
    is_boss: bool


class QuestDetailResponseData(BaseModel):
    quest: QuestDTO
    days: list[QuestDaySummaryDTO]


class ReviewAssetDTO(BaseModel):
    asset_id: str
    asset_type: str
    asset_name: str


class NextWeekSuggestionDTO(BaseModel):
    quest_title: str
    description: str


class ReviewDataDTO(BaseModel):
    completed_count: int
    total_days: int
    outputs: list[str]
    assets: list[ReviewAssetDTO]
    observations: list[str]
    boss_summary: str
    next_week_suggestion: NextWeekSuggestionDTO


class ReviewQuestSummaryDTO(BaseModel):
    quest_id: str
    status: str
    current_day: int


class ReviewResponseData(BaseModel):
    review: ReviewDataDTO
    quest: ReviewQuestSummaryDTO


class ReviewLLMSchema(BaseModel):
    outputs: list[str]
    observations: list[str]
    boss_summary: str
    next_week_suggestion: NextWeekSuggestionDTO


class NextWeekRequest(BaseModel):
    accept_suggestion: bool = True


class NextWeekResponseData(BaseModel):
    goal_id: str
    redirect: str
    message: str


class PauseQuestResponseData(BaseModel):
    quest_id: str
    status: str


class QuestPreviewLLMSchema(BaseModel):
    """LLM 结构化输出 Schema。"""

    quest_title: str
    success_condition: str
    total_estimated_time: str
    days: list[QuestPreviewDayDTO]

File: frontend/src/types/quest.ts
import type { AssetType, OutputType, QuestStatus } from './api'
import type { Goal } from './goal'

export interface QuestPreviewDay {
  day_index: number
  event_title: string
  event_description: string
  estimated_time: string
  meaning: string
  output_type: OutputType
  is_boss: boolean
}

export interface QuestPreview {
  quest_title: string
  success_condition: string
  total_estimated_time: string
  days: QuestPreviewDay[]
}

export interface Quest {
  quest_id: string
  goal_id: string
  quest_title: string
  success_condition: string
  total_days: number
  current_day: number
  status: QuestStatus
  created_at: string
  updated_at: string
}

export interface QuestDaySummary {
  day_index: number
  event_title: string
  estimated_time: string
  status: string
  is_boss: boolean
}

export interface ActiveQuestResponse {
  quest: Quest
  progress: { completed_days: number; total_days: number }
  assets_count: number
}

export interface QuestDetailResponse {
  quest: Quest
  days: QuestDaySummary[]
}

export interface ReviewAsset {
  asset_id: string
  asset_type: AssetType
  asset_name: string
}

export interface ReviewData {
  completed_count: number
  total_days: number
  outputs: string[]
  assets: ReviewAsset[]
  observations: string[]
  boss_summary: string
  next_week_suggestion: {
    quest_title: string
    description: string
  }
}

export interface ReviewResponse {
  review: ReviewData
  quest: Pick<Quest, 'quest_id' | 'status' | 'current_day'>
}

export interface NextWeekResponse {
  goal_id: string
  redirect: string
  message: string
}

export interface PauseQuestResponse {
  quest_id: string
  status: QuestStatus
}

export interface AcceptQuestPreview {
  quest_title: string
  success_condition: string
  days: QuestPreviewDay[]
}

export interface AcceptQuestResponse {
  quest: Quest
  today_event: import('./event').DailyEvent
}

export type { Goal }
